### PR TITLE
Fix Issue 18651 - ice: assert in glue.d:777 when building these three…

### PR DESCRIPTION
**Pull request by BorisCarvajal**
… trivial files

The problem was that some functions were not with sematic3 pass done because their symbol parent was deferred and `runDeferredSemantic3()` wasn't run. In this case were two auto generated functions (`__xopEquals, __xtoHash` that start only with semantic2) of some instantiation of `SortedRange`.

The test case on bugzilla uses Phobos and consumes 550MB of RAM, moreover, I think crafting another reduced case could be really time consuming, just in this example there were like ten thousand symbols deferred but still only two functions left over.

### CODE REVIEW

**Comment by Geod24** [comment]
The test in bugzilla does use phobos, but it doesn't so much but an import.
Personally I'd prefer to have a test case for this considering it's likely that such a regression creeps back, but I'll let others chime in.

**Comment by BorisCarvajal** [comment]
@Geod24 , indeed the test just imports two packages but the feature of concern (`-deps`) runs semantic3 on most if not all the Phobos modules, if that's not a problem I'll add it.

**Comment by Geod24** [comment]
@BorisCarvajal : I guess the ideal would be to come up with a test that depends on deferred sema3 being run but not Phobos. How time consuming would that be ?

**Comment by BorisCarvajal** [comment]
I don't know. But I've uploaded the original test case, it's better than nothing.

**Review by MoonlightSentinel — CHANGES_REQUESTED** [review]
Please let me have a look before merging, this test shouldn't pass ATM ....

EDIT: 
`TEST_OUTPUT` isn't enforced because this PR targets stable and the `d_do_test` extensions weren't merged yet. So we should probably have a master -> stable merge before proceeding with this PR.

**Comment by BorisCarvajal** [comment]
@MoonlightSentinel, I'll use -deps=file to make the compilation silent, the output is long and can vary.

**Comment by Geod24** [comment]
> So we should probably have a master -> stable merge before proceeding with this PR.

`stable` => master is a free for all, but master => stable happens at specific time.
I'm a bit fuzzy on the details now, but usually only Martin does it.

**Comment by MoonlightSentinel** [comment]
Dustmite found an interesting reduction:

```
// a.d:
import stdx.datetime;
```

```
// b.d:
import c;
```

```
// c.d:
import stdx.algorithm;

const var = sort(string[].init);
```

```
// stdx/algorithm.d:
struct SortedRange(Range)
{
    Range _input;
}

auto sort(Range)(Range)
{
    return SortedRange!Range();
}

auto uniq(Range)(Range)
{
    return SortedRange!Range();
}

```

```
// stdx/datetime.d:
void parseTZConversions()
{
    import algorithm; // Does not exist.

    string[] value;
    value.sort.uniq;
}
```

```
dmd -deps a.d b.d
```
This PR fixes the segfault but does not recognize the invalid import in `parseTZConversions`.

**Comment by BorisCarvajal** [comment]
Interesting reduction by the mighty Dustmite.
But doesn't look like an invalid import.

**Comment by MoonlightSentinel** [comment]
`parseTZConversions` imports `algorithm`, not `stdx.algorithm` from `stdx/algorithm.d`.

**Comment by BorisCarvajal** [comment]
@MoonlightSentinel , DMD loads the import just fine.
`stdx/algorithm.d` has no explicit module declaration, hence `datetime.d` from the same directory can import it with its plain name `algorithm`.
Whether you should use the prefix `stdx.` or not seems irrelevant to this issue.

Thanks anyway, I'll update the test to the reduced code.

(Note: This is different from your recent issue report, in which case the import file doesn't exist at all)

**Comment by MoonlightSentinel** [comment]
> Whether you should use the prefix `stdx.` or not seems irrelevant to this issue.

Agreed, itis just odd and feels like it should not work.

(Note: This is different from your recent issue report, in which case the import file doesn't exist at all)

Yeah, that was the first failed reduction.

File: test/compilable/test18651a.d
Comment (MoonlightSentinel) [inline test/compilable/test18651a.d:1]:
```suggestion
// REQUIRED_ARGS: -deps=${RESULTS_DIR}/compilable/test18651a.deps
```

File: test/compilable/test18651a.d
Comment (MoonlightSentinel) [inline test/compilable/test18651a.d:5]:
```suggestion
import imports.test18651_datetime;
```

**Review by MoonlightSentinel — CHANGES_REQUESTED** [review]
Please use explicit imports instead of an additional import path (it avoids confusion about the location of these files).

You could also create an additional directory (`imports/test18651`) for all addtional files used by this test.

File: test/compilable/test18651a.d
Comment (BorisCarvajal) [reply test/compilable/test18651a.d]: Done.

**Review by MoonlightSentinel — APPROVED** [review]
LGTM

**Review by Geod24 — APPROVED** [review]
Amazing work, thanks

**Pull request merged**
